python/pyhdk/_common.pyx

```diff
@@ -184,6 +184,35 @@ cdef class TypeInfo:
   def __repr__(self):
     return self.c_type_info.toString()
 
+cdef class Config:
+  def setCpuProp(self, cpu_prop):
+    self.c_config.get().exec.heterogeneous.forced_cpu_proportion=cpu_prop
+    self.c_config.get().exec.heterogeneous.forced_gpu_proportion=100-cpu_prop
+
+  def setGpuProp(self, gpu_prop):
+    self.c_config.get().exec.heterogeneous.forced_gpu_proportion=gpu_prop
+    self.c_config.get().exec.heterogeneous.forced_cpu_proportion=100-gpu_prop
+
+  def getCpuProp(self):
+    return self.c_config.get().exec.heterogeneous.forced_cpu_proportion
+
+  def getGpuProp(self):
+    return self.c_config.get().exec.heterogeneous.forced_gpu_proportion
+
+  def setHeterogenExec(self, on_off):
+    self.c_config.get().exec.heterogeneous.enable_heterogeneous_execution=on_off
+
+  def setForcedHeterogenDist(self, on_off):
+    self.c_config.get().exec.heterogeneous.forced_heterogeneous_distribution=on_off
+
+  def isHeterogen(self):
+    return self.c_config.get().exec.heterogeneous.enable_heterogeneous_execution
+
+  def isForcedDist(self):
+    return self.c_config.get().exec.heterogeneous.forced_heterogeneous_distribution
+
+  def setMultifragExec(self, on_off):
+    self.c_config.get().exec.heterogeneous.enable_multifrag_heterogeneous_execution=on_off
```

Comment:
Can we leave the interface unchanged and just provide these to the `buildConfig`?

Reply:
I meant all the methods :)

Reply:
Exposing the config gives up control over it, which seems undesirable. So, the idea was to hide it from the user and be able to validate it on the HDK side.

Reply:
Agree. You can pass required flags to `pyhdk.init` to get the desired config. If you want to change config between queries, then it's a sign we should move/add some flags to the `ExecutionOptions`.

Reply:
Having these knobs in the execution options makes sense to me. Changing the config between queries, on the contrary, assumes the config object is shared which is an implementation detail, hence, undesirable. I'd suggest either moving these options from config to `ExecutionOptions` at all or at least mirror them in the execution options and adjust the `fromConfig` method appropriately. As to the notebook, it can then control the data distribution directly through `execute` / `run` API, no reinitialization required.